components/script/dom/element.rs

```diff
@@ -348,60 +348,50 @@ impl Element {
     }
 
     // https://drafts.csswg.org/cssom-view/#css-layout-box
-    //
-    // We'll have no content box if there's no fragment for the node, and we use
-    // bounding_content_box, for simplicity, to detect this (rather than making a more specific
-    // query to the layout thread).
     fn has_css_layout_box(&self) -> bool {
-        self.upcast::<Node>().bounding_content_box().is_some()
+        let style = self.upcast::<Node>().style();
+        style.map_or(false, |s| !s.get_box().clone_display().is_none())
```

Comment:
Please document why it's accurate? (because we won't return styles for elements in display: none subtrees)